trlx/model/nn/ilql_models.py

```diff
@@ -147,35 +149,35 @@ def loss(self, batch):
                     actions.reshape(-1),
                     reduction="none",
                 ).reshape(bsize, ntokens - 1)
-                * isterminal
+                * terminal_mask
             ).sum() / n_nonterminal
             loss_cql_q2 = (
                 F.cross_entropy(
                     qs[1][:, :-1].reshape(-1, dsize),
                     actions.reshape(-1),
                     reduction="none",
                 ).reshape(bsize, ntokens - 1)
-                * isterminal
+                * terminal_mask
             ).sum() / n_nonterminal
             loss_cql = loss_cql_q1 + loss_cql_q2
         else:
             loss_cql = (
                 F.cross_entropy(
                     qs[:, :-1].reshape(-1, dsize), actions.reshape(-1), reduction="none"
                 ).reshape(bsize, ntokens - 1)
-                * isterminal
+                * terminal_mask
             ).sum() / n_nonterminal
 
         loss_awac = (
             F.cross_entropy(
                 logits[:, :-1].reshape(-1, dsize), actions.reshape(-1), reduction="none"
             ).reshape(bsize, ntokens - 1)
-            * isterminal
+            * terminal_mask
         ).sum() / n_nonterminal
 
         loss = loss_q + loss_v + self.cql_scale * loss_cql + self.awac_scale * loss_awac
```

Comment:
Do you think it makes more sense to put the loss in the nn.module classes or the accelerator trainer classes? I thought we would want loss defined in the accelerator classes but I'm open to something different if you have a strong opinion?

Reply:
I don't have a strong opionion, but forward and generate are also specific here and have to be decoupled and I'm not sure there is a reason for that just yet